pennylane/gradients/general_shift_rules.py

```diff
@@ -391,16 +391,30 @@ def generate_shifted_tapes(tape, index, shifts, multipliers=None):
             The length should match the one of ``shifts``. Each multiplier scales the
             corresponding gate parameter before the shift is applied. If not provided, the
             parameters will not be scaled.
+        broadcast (bool): Whether or not to use broadcasting to create a single tape
+            with the shifted parameters.
 
     Returns:
         list[QuantumTape]: List of quantum tapes. In each tape the parameter indicated
             by ``index`` has been shifted by the values in ``shifts``. The number of tapes
-            matches the lenth of ``shifts`` and ``multipliers`` (if provided).
+            matches the length of ``shifts`` and ``multipliers`` (if provided).
+            If ``broadcast=True`` was used, the list contains a single broadcasted tape
+            with all shifts distributed over the broadcasting dimension. In this case,
+            the ``batch_size`` of the returned tape matches the length of ``shifts``.
     """
     params = list(tape.get_parameters())
     if multipliers is None:
         multipliers = np.ones_like(shifts)
 
+    if broadcast:
+        new_params = params.copy()
+        new_params[index] = new_params[index] * qml.math.convert_like(
+            multipliers, new_params[index]
+        ) + qml.math.convert_like(shifts, new_params[index])
+        shifted_tape = tape.copy(copy_operations=True)
+        shifted_tape.set_parameters(new_params)

```

Comment:
These steps seem identical to those on lines 421-426 (unless I missed something). How about creating an auxiliary function that is used in both places to help future maintainability?

Reply:
Very nice idea, looks :100: now :)